### PR TITLE
Added create endpoints

### LINKED ISSUE #964: Add create endpoints to server
Add an endpoint to create SEIs or CAs via the REST API.

### PR DESCRIPTION
**Pull request by JAmmermann-DLR**
Added three create endpoints and test cases for:
- RootSei in ModelAccessResource
- Sei in StructuralElementInstanceResource
- Ca in CategoryAssignmentResource

Also added some minor improvement in the server errors.

@SaMuellerDLR the only think we don't handle at the moment are composed beans. For example what if one wants to create a ca inside a composed bean?

Closes #964

### CODE REVIEW

File: de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/ModelAccessResource.java [hunk 1]
```diff
@@ -251,4 +294,28 @@ public Response getConcepts(@DefaultValue("true") @QueryParam(QP_ONLY_ACTIVE_CON
...
+	 */
+	public static String createSeiFromFqn(String fullQualifiedName, EObject owner, VirSatTransactionalEditingDomain editingDomain) {
+		ActiveConceptHelper helper = new ActiveConceptHelper(editingDomain.getResourceSet().getRepository());
+		
+		StructuralElementInstance newSei = StructuralFactory.eINSTANCE.createStructuralElementInstance();
+		
+		// Add the correct type using the fqn
+		int idx = fullQualifiedName.lastIndexOf(".");
+		Concept concept = helper.getConcept(fullQualifiedName.substring(0, idx));
+		StructuralElement se = ActiveConceptHelper.getStructuralElement(concept, fullQualifiedName.substring(idx + 1));
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/ModelAccessResource.java:313]: I feel like the logic of geting the StructuralElement for an FQN should be part of the ActiveConceptHelper so we don't have those details in the server logic.

File: de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/modelaccess/CategoryAssignmentResource.java [hunk 2]
```diff
@@ -114,6 +122,58 @@ public Response putCa(@ApiParam(value = "CA to put", required = true) ABeanCateg
...
+			
+			StructuralElementInstance parentSei = RepositoryUtility.findSei(parentUuid, repository);
+			
+			if (parentSei == null) {
+				return ApiErrorHelper.createNotFoundErrorResponse();
+			}
+			
+			ActiveConceptHelper helper = new ActiveConceptHelper(ed.getResourceSet().getRepository());
+			
+			CategoryAssignment newCa = CategoriesFactory.eINSTANCE.createCategoryAssignment();
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/modelaccess/CategoryAssignmentResource.java:159]: Maybe a minor simplification is possible using the CategoryInstantiator. What is the current default name of the instantiated category assignment? Null?

File: de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/ModelAccessResource.java [hunk 3]
```diff
@@ -251,4 +294,28 @@ public Response getConcepts(@DefaultValue("true") @QueryParam(QP_ONLY_ACTIVE_CON
...
+	 * Create a new Sei typed by the Se identified by the fullQualifiedName
+	 * @param fullQualifiedName of the sei type (se)
+	 * @param owner of the sei (either another sei or the repository for root seis)
+	 * @param editingDomain
+	 * @return uuid of the created sei
+	 */
+	public static String createSeiFromFqn(String fullQualifiedName, EObject owner, VirSatTransactionalEditingDomain editingDomain) {
+		ActiveConceptHelper helper = new ActiveConceptHelper(editingDomain.getResourceSet().getRepository());
+		
+		StructuralElementInstance newSei = StructuralFactory.eINSTANCE.createStructuralElementInstance();
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/ModelAccessResource.java:308]: StructuralInstantiator could be used here to simplify instantiation logic (creating sei + setting type + setting default name)

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
Great job as usual. I got some minor comments where I believe that there is too much logic in the server that should be in a model plugin and just get called from the server classes.

**Review by maurpa — APPROVED** [review]

**Comment by JAmmermann-DLR** [comment]
Falsely closed via #985

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Pull request merged**